python/vision_explanation_methods/error_labeling/error_labeling.py

```diff
@@ -0,0 +1,229 @@
+        :param classes: Class names as a list of strings.
+            The order of the class names should match that of the model
+            output. Only required if explaining classifier.
+        :type classes: list
+        :param image_mode: The mode to open the image in.
+            See pillow documentation for all modes:
+            https://pillow.readthedocs.io/en/stable/handbook/concepts.html
+        :type image_mode: str
+        """
+        self._is_run = False
+        self._is_added = False
+        self._task_type = task_type
+        self._pred_y = pred_y
+        self._true_y = true_y
+        self._iou_threshold = iou_threshold
+        self._match_matrix = np.full((len(self._true_y), len(self._pred_y)),
+                                     None)
+
+    def compute(self, **kwargs):
+        """Compute the error analysis data.
+
+        Note: if a row does not have a match, that means that there is a
+        missing gt detection
+
+        :param kwargs: The keyword arguments to pass to the compute method.
+            Note that this method does not take any arguments currently.
+        :type kwargs: dict
+        """
+        # save original ordering of predictions
+        original_indices = [i for i, _ in sorted(enumerate(self._pred_y),
+                                                 key=lambda x: x[1][-1],
+                                                 reverse=True)]
+
+        # sort predictions by decreasing conf score
+        # this is to stay consistent with NMS and MAP algorithms
+        sorted_list = sorted(self._pred_y, key=lambda x: x[-1], reverse=True)
+
+        if self._true_y == []:
+            self._match_matrix = [[ErrorLabelType.BACKGROUND]
+                                  for i in range(0, len(self._pred_y))]
+            return
+
+        for gt_index, gt in enumerate(self._true_y):
+            for detect_index, detect in enumerate(sorted_list):
+                iou_score = torchvision.ops.box_iou(
+                    Tensor(detect[1:5]).unsqueeze(0).view(-1, 4),
+                    Tensor(gt[1:5]).unsqueeze(0).view(-1, 4))
+
+                if iou_score.item() == 0.0:
+                    # if iou is 0, then prediction is detecting the background
+                    self._match_matrix[gt_index][detect_index] = (
+                        ErrorLabelType.BACKGROUND)
+                    continue
+                if (self._iou_threshold <= iou_score):
+                    # the detection and ground truth bb's are overlapping
+                    if detect[0] != gt[0]:
+                        # the bb's line up, but labels do not
+                        self._match_matrix[gt_index][detect_index] = (
+                            ErrorLabelType.CLASS_NAME)
+                        continue
+                    elif (ErrorLabelType.MATCH in
+                          self._match_matrix[gt_index]):
+                        # class name and bb correct, but there is already a
```

Comment:
```suggestion
                        # class name and bbox correct, but there is already a
```
Nit. Maybe replace bb with bbox in the file.